src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -345,8 +346,10 @@ private CallbackResponse aboutToSubmit(CallbackParams callbackParams, boolean v1
 
         MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
 
-        if (V_2.equals(callbackParams.getVersion()) && caseData.isRejectDefendantPaymentPlanNo()) {
+        if (V_2.equals(callbackParams.getVersion()) && caseData.hasApplicantRejectedRepaymentPlan()) {
             response.state(CaseState.PROCEEDS_IN_HERITAGE_SYSTEM.name());
+        } else if (V_2.equals(callbackParams.getVersion()) && caseData.isSettlePartAdmitClaimYes()) {
```

Comment:
perhaps this needs to be all under V2? even though it will still be ugly. and then have a method that would incorporporate both states

Reply:
Updated from suggestion